src/databricks/labs/ucx/progress/tables.py

```diff
@@ -54,15 +53,30 @@ def append_inventory_snapshot(self, snapshot: Iterable[Table]) -> None:
         # The mode is 'append'. This is documented as conflict-free.
         self._sql_backend.save_table(escape_sql_identifier(self.full_name), history_records, Historical, mode="append")
 
+    @cached_property
+    def _used_hive_tables(self) -> defaultdict[str, list[UsedTable]]:
```

Comment:
Why return a defaultdict? You can use defaultdict and return a dict. This is more common in our code.

Reply:
I thought that was necessary, it is indeed not needed, I changed it, thanks